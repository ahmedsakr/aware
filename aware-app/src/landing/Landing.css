#landing-canvas {
    flex-direction: column;
    justify-content: center;
    background-image: url('/shrek.jpg');
}

label {
    color: #a7afb2;
}

#landing-textfield {
    background-color: #18202c;
    width: 100%;
    padding: 8px 10px;
    display: inline-block;
    border: 1px solid #ccc;
    border-color: #18202c;
    border-radius: 7px;
    color: #a7afb2;
}

button {
    background-color: #ffc000;
    color: black;
    padding: 10px 10px;
    border: none;
    width: 100%;
    border-radius: 7px;
    margin-top: 15px;
}