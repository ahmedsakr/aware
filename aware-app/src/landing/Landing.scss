/*
 * Landing styling
 *
 * Defines layout fundamentals for the landing page.
 */

$landing_textfield_color: #18202c;
$label_font_color: #a7afb2;
$button_color: #ffc000;
$forgot_password_font_color: #008080;
$backround_image_url: '/shrek.jpg';

#landing-canvas {
    flex-direction: column;
    justify-content: center;
    background-image: url($backround_image_url);
  }
  
  label {
    color: $label_font_color;
  }
  
  .landing-textfield {
    background-color: $landing_textfield_color;
    width: 100%;
    padding: 8px 10px;
    display: inline-block;
    border: 1px solid #ccc;
    border-color: $landing_textfield_color;
    border-radius: 7px;
    color: $label_font_color;
  }
  
  button {
    background-color: $button_color;
    color: black;
    padding: 10px 10px;
    border: none;
    width: 100%;
    border-radius: 7px;
    margin-top: 15px;
  }
  
  label > span, #forgot-password {
    cursor: pointer;
    color: $forgot_password_font_color;
    text-decoration: underline;
  }
  
  label > span:hover, #forgot-password {
    text-decoration: none;
    text-shadow: 1px 1px 1px #555;
  }