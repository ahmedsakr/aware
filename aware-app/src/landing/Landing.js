import React, { Component } from 'react';
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
import './Landing.css'

import Login from './login/Login'
import Register from './register/Register'

class Landing extends Component {
  constructor() {
    super()

    this.state = {
      component: Register
    }
  }

  render() {
    var Component;
    if (this.state.component === Login) {
      Component = Login;
    } else {
      Component = Register;
    }

    return (
      <div className="container-fluid aware-container">
        <div id="landing-canvas" className="aware-column row">
          <div id="test-flex" className="col-3 offset-3">
            <Component switch = {this.switchView} loadMessenger = {this.props.loadMessenger}/>
          </div>
        </div>
      </div>
    );
  }

  switchView = () => {
    if (this.state.component === Login) {
      this.setState({
        component: Register
      })
    } else {
      this.setState({
        component: Login
      })
    }
  }
}

export default Landing;
