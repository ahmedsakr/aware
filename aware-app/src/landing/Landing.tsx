import React from 'react';
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
import './Landing.scss'

import Login from './login/Login'
import Register from './register/Register'

type LandingProps = {
    loadMessenger: (username: string) => void;
    socket: SocketIOClient.Socket
};

type LandingState = {
    component: string,
    username: string | null
};

export default class Landing extends React.Component<LandingProps, LandingState> {
    constructor(props: LandingProps) {
        super(props)

        this.state = {
            component: "login",
            username: null
        }

        this.setUsername.bind(this);
    }

    componentDidMount(): void {
        if (this.props.socket) {
            this.props.socket.on('login-request', (result: boolean) => {
                if (result) {
                    this.props.loadMessenger(this.state.username as string);
                } else {
                    alert("Invalid username or password.");
                }
            })

            this.props.socket.on('register-request', (result: boolean) => {
                if (result) {
                    alert("Registration successful!");
                    this.switchView();
                } else {
                    alert("Registration failure");
                }
            });
        }
    }

    displayCurrentForm = () => {
        if (this.state.component === "login") {

            return (
                <Login
                    socket={this.props.socket}
                    switch={this.switchView}
                    setUsername={this.setUsername} />
            );

        } else {

            return (
                <Register
                    socket={this.props.socket}
                    switch={this.switchView}
                    setUsername={this.setUsername} />
            );

        }
    }

    switchView = () => {
        let nextForm = this.state.component === "login" ? "register" : "login";
        this.setState({ component: nextForm });
    }

    setUsername = (username: string) => {
        this.setState({
            username: username
        });
    }

    render() {
        return (
            <div className="container-fluid aware-container">
                <div id="landing-canvas" className="aware-column row">
                    <div className="col-3 offset-3">
                        {this.displayCurrentForm()}
                    </div>
                </div>
            </div>
        );
    }
}
