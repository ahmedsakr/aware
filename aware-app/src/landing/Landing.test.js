import React from 'react';
import { shallow } from 'enzyme';

import Landing from './Landing';

describe('Landing', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<Landing debug />);
  
    expect(component).toMatchSnapshot();
  });
});