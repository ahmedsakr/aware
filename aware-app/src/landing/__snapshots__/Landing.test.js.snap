// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render correctly in "debug" mode 1`] = `
<div
  className="container-fluid aware-container"
>
  <div
    className="aware-column row"
    id="landing-canvas"
  >
    <div
      className="col-3 offset-3"
      id="test-flex"
    >
      <Register
        switch={[Function]}
      />
    </div>
  </div>
</div>
`;
